rust/lance/src/dataset/write/delete.rs

```diff
@@ -183,7 +185,9 @@ impl RetryExecutor for DeleteJob {
                         .iter()
                         .map(|f| f.id() as u64)
                         .collect();
-                    (Vec::new(), deleted_fragment_ids)
+                    // When deleting everything, we don't have specific row addresses,
+                    // but we can create an empty RowIdTreeMap since all fragments are deleted
+                    (Vec::new(), deleted_fragment_ids, RowIdTreeMap::new())
```

Comment:
I think the logic should be right here based on the `conflict_resolver` logic, but can we add a test to confirm?

Reply:
Good catch. Added a test and modified this not to return an `affected_rows` here.